src/model64_internal.h

```diff
@@ -34,21 +34,73 @@ typedef struct primitive_s {
     void *indices;                  ///< Pointer to the first index value. If NULL, indices are not used
 } primitive_t;
 
-/** @brief A mesh that is made up of multiple primitives (part of #model64_t) */
+/** @brief Transform of a node of a model */
+typedef struct node_transform_s {
+    float pos[3];       ///< Position of a node
+    float rot[4];       ///< Rotation of a node (quaternion)
+    float scale[3];     ///< Scale of a node
+    float mtx[16];      ///< Matrix of a node
+} node_transform_t;
+
+/** @brief Transform state of a node of a model */
+
+typedef struct node_transform_state_s {
+    node_transform_t transform;     ///< Current transform state for a node
+    float world_mtx[16];            ///< World matrix for a node
```

Comment:
Is the naming "world_mtx" actually accurate? I guess it's in "world space" relative to the scene that is described by the model file. But one could also interpret that to mean world space in the application as a whole (even though that doesn't make much sense). Maybe "model_mtx" instead to avoid confusion?

Reply:
Would that also change the name of model64_get_node_world_mtx.

Reply:
It would, but now I'm second guessing my proposed naming. Maybe "world_mtx" is fine for now.